File: reciclaje-ia/frontend/src/utils/validaciones.js
export const validarCedulaEcuatoriana = (cedula) => {
  const cedulaLimpia = cedula.replace(/\D/g, '');
  
  if (cedulaLimpia.length !== 10) {
    return { valida: false, mensaje: 'La cédula debe tener 10 dígitos' };
  }

  if (/^0{10}$/.test(cedulaLimpia)) {
    return { valida: false, mensaje: 'Cédula inválida' };
  }

  const provincia = parseInt(cedulaLimpia.substring(0, 2));
  if (provincia < 1 || provincia > 24) {
    return { valida: false, mensaje: 'Provincia inválida' };
  }

  const digitos = cedulaLimpia.split('').map(Number);
  const primerDigito = digitos[9];
  let suma = 0;

  for (let i = 0; i < 9; i++) {
    let multiplicador = i % 2 === 0 ? 2 : 1;
    let valor = digitos[i] * multiplicador;
    if (valor >= 10) valor -= 9;
    suma += valor;
  }

  const digitoVerificador = (10 - (suma % 10)) % 10;

  if (digitoVerificador !== primerDigito) {
    return { valida: false, mensaje: 'Cédula inválida - dígito verificador incorrecto' };
  }

  return { valida: true, mensaje: 'Cédula válida' };
};

export const validarEmail = (email) => {
  const regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return regex.test(email);
};

export const validarPassword = (password) => {
  if (password.length < 6) {
    return { valida: false, mensaje: 'La contraseña debe tener al menos 6 caracteres' };
  }
  return { valida: true, mensaje: 'Contraseña válida' };
};

File: reciclaje-ia/backend/app/utils/validaciones.py
def validar_cedula_ecuatoriana(cedula):
    cedula = cedula.replace(' ', '').replace('-', '')
    
    if len(cedula) != 10:
        return {'valida': False, 'mensaje': 'La cédula debe tener 10 dígitos'}
    if not cedula.isdigit():
        return {'valida': False, 'mensaje': 'La cédula debe contener solo números'}
    
    provincia = int(cedula[:2])
    if provincia < 1 or provincia > 24:
        return {'valida': False, 'mensaje': 'Provincia inválida'}
    
    digitos = [int(d) for d in cedula]
    ultimo_digito = digitos[-1]
    suma = 0
    
    for i in range(9):
        valor = digitos[i]
        if i % 2 == 0:
            valor *= 2
            if valor >= 10:
                valor -= 9
        suma += valor
    
    digito_verificador = (10 - (suma % 10)) % 10
    
    if digito_verificador != ultimo_digito:
        return {'valida': False, 'mensaje': 'Cédula inválida'}
    
    return {'valida': True, 'mensaje': 'Cédula válida'}